src/main-oe/sgxlkl_run_oe.c

```diff
@@ -2032,24 +2037,26 @@ int main(int argc, char* argv[], char* envp[])
         pthread_setname_np(sgxlkl_threads[i], "ENCLAVE");
     }
 
-    long exit_status = 0;
-
-    for (int i = 0; i < econf->ethreads; i++)
-    {
-        pthread_join(sgxlkl_threads[i], &return_value);
-        if (i == 0)
-        {
-            exit_status = (long)return_value;
-        }
-    }
-    sgxlkl_host_verbose_raw("\n");
-
-    if (oe_enclave)
-    {
-        sgxlkl_host_verbose("oe_terminate_enclave... ");
-        oe_terminate_enclave(oe_enclave);
-        sgxlkl_host_verbose_raw("done\n");
-    }
+    pthread_mutex_lock(&first_ethread_exited_mtx);
+    ret =
+        pthread_cond_wait(&first_ethread_exited_cv, &first_ethread_exited_mtx);
+    if (ret != 0)
+        sgxlkl_host_fail("Failed to wait for enclave to finish: ret=%i\n", ret);
+    long exit_status = (long)enclave_return_status;
+    pthread_mutex_unlock(&first_ethread_exited_mtx);
+
+    /**
+     * This fails with an error when we try to terminate the enclave while
+     * ethreads are still executing (?). We need to figure out a way to shutdown
+     * the enclave under coperative scheduling, i.e., when an ethread may be
+     * stuck executing an application busy-loop.
+     */
+    // if (oe_enclave)
```

Comment:
What about this commented out code?

Reply:
Same point as above. We should have a discussion if it is right to remove this. I also don't understand why the shutdown doesn't work anymore.